packages/mongodb-runner/src/util.ts

```diff
@@ -6,3 +6,16 @@ export const uuid = () => new BSON.UUID().toHexString(true);
 export const sleep = (ms: number): Promise<void> =>
   new Promise((r) => setTimeout(r, ms));
 export const range = (n: number): number[] => [...Array(n).keys()];
+export async function parallelForEach<T>(
+  iterable: AsyncIterable<T>,
+  fn: (arg0: T) => Promise<void> | void
+) {
+  const items = iterable[Symbol.asyncIterator]();
+
+  const result = [];
+  for (let item = await items.next(); !item.done; item = await items.next()) {
+    result.push(fn(item.value));
+  }
+
+  return await Promise.allSettled(result);
```

Comment:
This effectively try/catches everything, is that intended here, especially since it’s a generic utility?

Reply:
Yes.  The result is returned so it's accessable to callers, but this helper applies `fn` without waiting and awaits them all at the end.  The intention is to wait for all `fn`s to finish before returning, regardless of success or failure

I wrote this because `prune` was slow when there were many dead clusters, because `prune` waits `serverSelectionTimeoutMS` for each cluster.  I set `serverSelectionTimeoutMS` to 2s to speed `prune` up, but it's still slow when there are more than one or two clusters if each cluster is connected to sequentially

Reply:
Yeah, totally valid use case and implementation :+1: I’d maybe consider incorporating something that calls out the lack of exception propagation to the function name or leave a comment about it, but the code itself is just fine